File: frontend/src/types/interview.ts
// 注意：后端 API 返回 snake_case，类型声明保持一致

export type QuestionSource = 'resume' | 'knowledge_base' | 'mixed'
export type QuestionSourceType = 'resume_experience' | 'resume_project' | 'knowledge_base'
export type InterviewStatus = 'created' | 'in_progress' | 'paused' | 'completed' | 'cancelled'

export interface InterviewSession {
  id: string
  candidate_id?: string
  resume_id?: string
  langgraph_thread_id: string
  status: InterviewStatus
  question_source?: QuestionSource
  total_questions: number
  completed_questions: number
  total_score?: number
  started_at?: string
  ended_at?: string
  created_at: string
  updated_at: string
}

export interface InterviewSessionCreate {
  candidate_id: string
  resume_id: string
  question_source: QuestionSource
  total_questions: number
}

export interface GeneratedQuestion {
  id: string
  interview_session_id: string
  source_type: QuestionSourceType
  source_kb_item_id?: string
  source_resume_context?: string
  question_text: string
  ai_reference_answer?: string
  question_order: number
  status: string
  created_at: string
}

export interface AnswerSubmit {
  question_id: string
  transcript_text: string
  audio_minio_key?: string
  audio_duration_sec?: number
}

export interface Answer {
  id: string
  question_id: string
  transcript_text: string
  audio_minio_key?: string
  audio_duration_sec?: number
  created_at: string
}

export interface AnswerAnalysis {
  id: string
  answer_id: string
  question_id: string
  eval_mode: 'rag_hybrid' | 'llm_judge'
  analysis_json: AnswerAnalysisData
  agent_model: string
  tokens_used?: number
  processing_ms?: number
  created_at: string
}

export interface AnswerAnalysisData {
  overall_score: number
  assessment: string
  strengths: string[]
  areas_for_improvement: string[]
  // LLM-as-a-Judge
  accuracy_score?: number
  accuracy_reasoning?: string
  completeness_score?: number
  completeness_reasoning?: string
  clarity_score?: number
  clarity_reasoning?: string
  technical_depth_score?: number
  technical_depth_reasoning?: string
  authenticity_flag?: string
  // RAG hybrid
  vector_similarity?: number
  covered_points?: string[]
  missing_points?: string[]
  retrieved_chunks?: Array<{ id: string; similarity: number; text: string }>
}

File: backend/app/schemas/interview.py
"""Interview-related Pydantic schemas."""

from datetime import datetime
from decimal import Decimal
from typing import Any, Dict, List, Literal, Optional
from uuid import UUID

from pydantic import BaseModel, Field


class InterviewSessionCreate(BaseModel):
    candidate_id: UUID
    resume_id: UUID
    question_source: Literal["resume", "knowledge_base", "mixed"]
    total_questions: int = Field(default=5, ge=1, le=20)


class InterviewSessionResponse(BaseModel):
    id: UUID
    candidate_id: Optional[UUID] = None
    resume_id: Optional[UUID] = None
    langgraph_thread_id: str
    status: str
    question_source: Optional[str] = None
    total_questions: int
    completed_questions: int
    total_score: Optional[Decimal] = None
    started_at: Optional[datetime] = None
    ended_at: Optional[datetime] = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class KBSourceConfig(BaseModel):
    """单个分类的题目配置。"""
    category_id: Optional[UUID] = None  # None = 不限分类
    count: int = Field(default=2, ge=0, le=20)


class QuestionGenerateRequest(BaseModel):
    source: Literal["resume", "knowledge_base", "mixed"] = "resume"
    count: int = Field(default=5, ge=1, le=20)
    # KB/mixed 模式：按分类指定题目数，为空则从全部分类随机
    kb_configs: List[KBSourceConfig] = Field(default_factory=list)
    # mixed 模式：resume 占比（0-100），如 60 表示 60% resume + 40% KB
    resume_ratio: int = Field(default=60, ge=0, le=100)
    # 题目池：这些题库条目的 ID 保证出现在面试题中（已向量化的 KB 条目）
    pool_ids: List[UUID] = Field(default_factory=list)


class QuestionResponse(BaseModel):
    id: UUID
    interview_session_id: UUID
    source_type: str
    source_kb_item_id: Optional[UUID] = None
    source_resume_context: Optional[str] = None
    question_text: str
    ai_reference_answer: Optional[str] = None
    question_order: int
    status: str
    created_at: datetime

    model_config = {"from_attributes": True}


class AnswerSubmitRequest(BaseModel):
    question_id: UUID
    transcript_text: str
    audio_minio_key: Optional[str] = None
    audio_duration_sec: Optional[int] = None


class AnswerResponse(BaseModel):
    id: UUID
    question_id: UUID
    transcript_text: str
    audio_minio_key: Optional[str] = None
    audio_duration_sec: Optional[int] = None
    created_at: datetime

    model_config = {"from_attributes": True}


class AnswerAnalysisResponse(BaseModel):
    id: UUID
    answer_id: UUID
    question_id: UUID
    eval_mode: str
    analysis_json: Dict[str, Any]
    agent_model: str
    tokens_used: Optional[int] = None
    processing_ms: Optional[int] = None
    created_at: datetime

    model_config = {"from_attributes": True}


class InterviewStateResponse(BaseModel):
    """Debug response showing full LangGraph state."""
    state: Dict[str, Any]
